argo/kfp-compiler/Makefile

```diff
@@ -13,17 +13,20 @@ penguin-staging:
 	cp ../../docs-gen/includes/quickstart/penguin_pipeline/*.py $(PENGUIN_STAGING)
 	cp integration/Dockerfile $(PENGUIN_STAGING)
 
-penguin-37: penguin-staging
-	docker build --build-arg PYTHON_VERSION=3.7.12 --build-arg TFX_VERSION=1.9.1 -t penguin:3.7 $(PENGUIN_STAGING)
+penguin-310: penguin-staging
+	docker build --build-arg PYTHON_VERSION=3.10.12 --build-arg TFX_VERSION=1.14.0 -t penguin:3.10 $(PENGUIN_STAGING)
 
 penguin-39: penguin-staging
 	docker build --build-arg PYTHON_VERSION=3.9.13 --build-arg TFX_VERSION=1.9.1 -t penguin:3.9 $(PENGUIN_STAGING)
 
-integration-test: docker-build penguin-37 penguin-39
+integration-test: docker-build penguin-310 penguin-39
 	$(eval TMP := $(shell mktemp -d))
+	echo '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}' > $(TMP)/pipeline_config.json
 	docker run -v $(TMP):/shared kfp-operator-argo-kfp-compiler /shared
-	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.7 --output_file /tmp/pipeline.yaml --pipeline_config '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}'
-	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}'
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json --execution_mode v1
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.10 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json  --execution_mode v1
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json --execution_mode v2
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.10 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json  --execution_mode v2
```

Comment:
Probably not something we can solve right now but in the future, this would benefit from defining a compatibility matrix and then iterating over it instead of duplicating this line

Reply:
i assume from the POV of OSS users' compatibility

Reply:
given that we test versions in the acceptance tests, we can just test one version here (v2)